clients/java/src/test/java/io/zeebe/client/OAuthCredentialsProviderTest.java

```diff
@@ -0,0 +1,190 @@
+        .containsEntry("client_secret", "secret")
+        .containsEntry("audience", "endpoint");
+  }
+
+  @Test
+  public void shouldFailWithNoAudience() {
+    // when/then
+    assertThatThrownBy(
+            () ->
+                new OAuthCredentialsProviderBuilder()
+                    .clientId("a")
+                    .clientSecret("b")
+                    .authorizationServerUrl("http://some.url")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageEndingWith(
+            String.format(OAuthCredentialsProviderBuilder.INVALID_ARGUMENT_MSG, "audience"));
+  }
+
+  @Test
+  public void shouldFailWithNoClientId() {
+    // when/then
+    assertThatThrownBy(
+            () ->
+                new OAuthCredentialsProviderBuilder()
+                    .audience("a")
+                    .clientSecret("b")
+                    .authorizationServerUrl("http://some.url")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageEndingWith(
+            String.format(OAuthCredentialsProviderBuilder.INVALID_ARGUMENT_MSG, "client id"));
+  }
+
+  @Test
+  public void shouldFailWithNoClientSecret() {
+    // when/then
+    assertThatThrownBy(
+            () ->
+                new OAuthCredentialsProviderBuilder()
+                    .audience("a")
+                    .clientId("b")
+                    .authorizationServerUrl("http://some.url")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageEndingWith(
+            String.format(OAuthCredentialsProviderBuilder.INVALID_ARGUMENT_MSG, "client secret"));
+  }
+
+  @Test
+  public void shouldFailWithNoAuthServerUrl() {
+    // when/then
+    assertThatThrownBy(
+            () ->
+                new OAuthCredentialsProviderBuilder()
+                    .audience("a")
+                    .clientId("b")
+                    .clientSecret("c")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageEndingWith(
+            String.format(
+                OAuthCredentialsProviderBuilder.INVALID_ARGUMENT_MSG, "authorization server URL"));
+  }
+
+  @Test
+  public void shouldFailWithMalformedServerUrl() {
+    // when/then
+    assertThatThrownBy(
+            () ->
+                new OAuthCredentialsProviderBuilder()
+                    .audience("a")
+                    .clientId("b")
+                    .clientSecret("c")
+                    .authorizationServerUrl("someServerUrl")
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasCauseInstanceOf(MalformedURLException.class);
+  }
+
+  private HttpURLConnection mockAuthServerConnection(OAuthCredentialsProviderBuilder builder)
+      throws IOException {
+    final String jsonResponse = "{\"access_token\":\"someToken\",\"token_type\":\"Bearer\"}";
```

Comment:
Seems kind of a pain to stub HTTP calls. Once we have more tests, it might be worth looking into [WireMock](http://wiremock.org/docs/getting-started/) or [OkHTTP MockWebServer](https://github.com/square/okhttp/tree/master/mockwebserver)

I'm not sure about it now because for one or two tests, WireMock might be too hefty a dependency to justify its use...